src/server/middlewares/resolveChainIds.js

```diff
@@ -0,0 +1,80 @@
+import httpStatus from 'http-status';
+
+import APIError from '~/server/helpers/errors';
+import Answer from '~/server/models/answer';
+import Question from '~/server/models/question';
+import logger from '~/server/helpers/logger';
+
+// Get answer chainIds which will be stored publicly on the blockchain
+export default async function resolveChainIdsMiddleware(req, res, next) {
+  const {
+    festivalAnswerIds,
+    artworkAnswerIds,
+    artworkQuestionId,
+    festivalQuestionId,
+  } = req.body;
+
+  try {
+    // @TODO: Make all of this into one single database query
+    const festivalAnswerChainIds = await Promise.all(
+      festivalAnswerIds.map(async (answerId) => {
+        const answer = await Answer.findByPk(answerId);
+
+        if (!answer) {
+          throw Error(`Could not find answer ${answerId} in database`);
+        }
+
+        // Festival answers can only be related to artworks
+        if (answer.propertyId) {
+          throw Error(`Answer ${answerId} is only for properties`);
+        }
+
+        return answer.chainId;
+      }),
+    );
+
+    const artworkAnswerChainIds = await Promise.all(
+      artworkAnswerIds.map(async (answerId) => {
+        const answer = await Answer.findByPk(answerId);
+
+        if (!answer) {
+          throw Error(`Could not find answer ${answerId} in database`);
+        }
+
+        // Artwork answers can only be related to properties
+        if (answer.artworkId) {
+          throw Error(`Answer ${answerId} is only for artworks`);
+        }
+
+        return answer.chainId;
+      }),
+    );
+
+    const artworkQuestionChainId = await Question.findByPk(
```

Comment:
these aren't secret, the client can just have the chain id

Reply:
Thought it could be good to resole the `chainId` for later use by the controller and the API design stays consistent for the client (you worry about `id`s and not a mix of different fields).